### PR TITLE
ENG-4567 Added map inheritance to EntThreadLocal

### PR DESCRIPTION
**Pull request by zonia3000**

### CODE REVIEW

File: cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentpagemapper/ContentPageMapperManager.java [hunk 1]
```diff
@@ -69,6 +70,10 @@ public String getPageCode(String contentId) {
 
 	@Override
 	public void updateFromPageChanged(PageChangedEvent event) {
+		if (_logger.isDebugEnabled()) {
```
Comment (entgigi) [inline cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentpagemapper/ContentPageMapperManager.java]: can we change from _logger to logger and use Slf4j or @Slf4j ?

File: engine/src/main/java/com/agiletec/aps/system/RequestContext.java [hunk 2]
```diff
@@ -53,14 +54,22 @@ public Object getExtraParam(String name) {
...
-	public void addExtraParam(String name, Object param) {
+	public void addExtraParam(String name, Serializable param) {
 		if (isCurrentFrameOrCurrentWidget(name)) {
 			EntThreadLocal.set(name, param);
 		} else {
 			this._extraParams.put(name, param);
 		}
 	}
 
+	public void setExecutorBeanContainer(ExecutorBeanContainer param) {
```
Comment (entgigi) [inline engine/src/main/java/com/agiletec/aps/system/RequestContext.java:71]: we should add an annotation @Deprecated ad a link to the problem / ticket to explain the problem / solution

File: engine/src/test/java/com/agiletec/aps/tags/HeadInfoOutputterTagTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    @Test
+    void shouldIncludeExistingInfo() throws Exception {
+        HeadInfoOutputterTag tag = new HeadInfoOutputterTag();
+        tag.setType(TYPE);
+        tag.setPageContext(pageContext);
+
+        Assertions.assertEquals(Tag.EVAL_BODY_INCLUDE, tag.doStartTag());
+        Assertions.assertEquals(IterationTag.EVAL_BODY_AGAIN, tag.doAfterBody());
+        Assertions.assertEquals("info2", tag.getCurrentInfo());
+        Assertions.assertEquals(Tag.SKIP_BODY, tag.doAfterBody());
```
Comment (entgigi) [inline engine/src/test/java/com/agiletec/aps/tags/HeadInfoOutputterTagTest.java:49]: in general as info for test we should not use constants from code but from tests, tests are more close to specification and requirements than code

**Review by entgigi — CHANGES_REQUESTED** [review]

**Review by entgigi — APPROVED** [review]
LGTM

File: cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentpagemapper/ContentPageMapperManager.java (hunk 1 above)
Comment (zonia3000) [reply cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/contentpagemapper/ContentPageMapperManager.java]: Done

File: engine/src/main/java/com/agiletec/aps/system/RequestContext.java (hunk 2 above)
Comment (zonia3000) [reply engine/src/main/java/com/agiletec/aps/system/RequestContext.java:71]: I added a warning here and in the ExecutorBeanContainer class. Please have a look, thanks.

**Pull request merged**